Add searchbp metrics to Performance Analyzer (#5390)
opensearch-project/documentation-website: _monitoring-your-cluster/pa/reference.md

   </tbody>
</table>

## Relevant dimensions: `NodeID`, `searchbp_mode`

<table>
  <thead style="text-align: left">
  <tr>
    <th>Metric</th>
    <th>Description</th>
  </tr>
 </thead>
 <tbody> 
    <tr>
    <td>searchbp_shard_stats_cancellationCount
    </td>

>     <td>The number of tasks marked for cancellation on the shard task.

Reviewer: "Shard task" or just "shard"?

>     <td>The number of tasks marked for cancellation on the shard task.

Reviewer: ```suggestion
    <td>The number of tasks marked for cancellation on the `shard_task`.
```

    </td>
  </tr>
  <tr>
    <td>SearchBP_Shard_Stats_LimitReachedCount
    </td>

>     <td>The number of times when the cancellable task total exceeded the set cancellation threshold at the shard task level.

Reviewer 2: ```suggestion
    <td>The number of times that the cancellable task total exceeded the set cancellation threshold at the shard task level.
```

    </td>
  </tr>
  <tr>
    <td>SearchBP_Shard_Stats_Resource_Heap_Usage_CancellationCount
    </td>
    <td>The number of tasks marked for cancellation because of excessive heap usage since the node last restarted at the shard task level.
    </td>
  </tr>
  <tr>
    <td>SearchBP_Shard_Stats_Resource_Heap_Usage_CurrentMax
    </td>
    <td>The maximum heap usage for tasks currently running at the shard task level.
    </td>
  </tr>
  <tr>
    <td>SearchBP_Shard_Stats_Resource_Heap_Usage_RollingAvg
    </td>

>     <td> The rolling average heap usage for the _n_ most recent tasks at the shard task level. The default value for _n_ is 100.

Reviewer 2: Should "100" be in code font?

>     <td> The rolling average heap usage for the _n_ most recent tasks at the shard task level. The default value for _n_ is 100.

Reviewer: ```suggestion
    <td> The rolling average heap usage for the _n_ most recent tasks at the shard task level. The default value for _n_ is `100`.
```

    </td>
  </tr>
  <tr>

>     <td>searchbp_shard_stats_resource_cpu_usage_cancellationCount

Reviewer: ```suggestion
    <td>SearchBP_Shard_Stats_Resource_CPU_Usage_CancellationCount
```

    </td>

>     <td>The number of tasks marked for cancellation because of excessive CPU usage since the node last restarted on the shard task.

Reviewer: ```suggestion
    <td>The number of tasks marked for cancellation because of excessive CPU usage since the last restart of the node containing the shard task.
```

    </td>
  </tr>
  <tr>
    <td>searchbp_shard_stats_resource_cpu_usage_currentMax
    </td>

>     <td>The maximum CPU time for all tasks currently running on the node on the shard task.

Reviewer: ```suggestion
    <td>The maximum CPU time for all tasks currently running on the node containing the shard task.
```

Reviewer: Does this retain the meaning?

    </td>
  </tr>
  <tr>
    <td>searchbp_shard_stats_resource_cpu_usage_currentAvg
    </td>

>     <td>The average CPU time for all tasks currently running on the node on the shard task.

Reviewer: ```suggestion
    <td>The average CPU time for all tasks currently running on the node containing the shard task.
```

Reviewer 3: ```suggestion
    <td>The average CPU time for all tasks currently running on the node at the shard level task.
```

Here the metric means that `cpu_usage` is for all shard level tasks running on the node.

Reviewer: Shard level tasks running on a single node?

    </td>
  </tr>
  <tr>
    <td>searchbp_shard_stats_resource_elaspedtime_usage_cancellationCount
    </td>

>     <td>The number of tasks marked for cancellation because of excessive elapsed time since the node last restarted on the shard task.

Reviewer: ```suggestion
    <td>The number of tasks marked for cancellation because of excessive time elapsed since the last restart of the node containing the shard task.
```

    </td>
  </tr>
  <tr>
    <td>searchbp_shard_stats_resource_elaspedtime_usage_currentMax
    </td>

>     <td>The maximum elapsed time for all tasks currently running on the node on the shard task.

Reviewer: ```suggestion
    <td>The maximum time elapsed for all tasks currently running on the node containing the shard task.
```

    </td>
  </tr>
  <tr>
    <td>searchbp_shard_stats_resource_elaspedtime_usage_currentAvg
    </td>

>     <td>The average elapsed time for all tasks currently running on the node on the shard task.

Reviewer: ```suggestion
    <td>The average time elapsed for all tasks currently running on the node containing the shard task.
```

    </td>
  </tr>
  <tr>
    <td>Searchbp_Task_Stats_CancellationCount
    </td>
    <td>The number of tasks marked for cancellation at the search task level.
    </td>
  </tr>
  <tr>
    <td>SearchBP_Task_Stats_LimitReachedCount
    </td>

>     <td>The number of times when the cancellable task total exceeded the set cancellation threshold at the search task level.

Reviewer 2: ```suggestion
    <td>The number of times that the cancellable task total exceeded the set cancellation threshold at the search task level.
```

    </td>
  </tr>
  <tr>
    <td>searchbp_task_stats_resource_heap_usage_cancellationCount
    </td>

>     <td>The number of tasks marked for cancellation because of excessive heap usage since the node last restarted on the search task level.

Reviewer: ```suggestion
    <td>The number of tasks marked for cancellation because of excessive heap usage since the last restart of the node containing the search task level.
```

    </td>
  </tr>
  <tr>
    <td>SearchBP_Task_Stats_Resource_Heap_Usage_CurrentMax
    </td>
    <td>The maximum heap usage for tasks currently running at the search task level.
    </td>
  </tr>
  <tr>
    <td>SearchBP_Task_Stats_Resource_Heap_Usage_RollingAvg
    </td>

>     <td> The rolling average heap usage for the _n_ most recent tasks on the search task level. The default value for _n_ is 10.

Reviewer 2: ```suggestion
    <td> The rolling average heap usage for the _n_ most recent tasks at the search task level. The default value for _n_ is 10.
```

>     <td> The rolling average heap usage for the _n_ most recent tasks on the search task level. The default value for _n_ is 10.

Reviewer 2: Should "10" be in code font?

    </td>
  </tr>
  <tr>
    <td>searchbp_task_stats_resource_cpu_usage_cancellationCount
    </td>

>     <td>The number of tasks marked for cancellation because of excessive CPU usage since the node last restarted on the search task level.

Reviewer: ```suggestion
    <td>The number of tasks marked for cancellation because of excessive CPU usage since the last restart of the node containing the search task level.
```

    </td>
  </tr>
  <tr>
    <td>searchbp_task_stats_resource_cpu_usage_currentMax
    </td>

>     <td>The maximum CPU time for all tasks currently running on the node on the search task level.

Reviewer: ```suggestion
    <td>The maximum CPU time for all tasks currently running on the node containing the search task level.
```

    </td>
  </tr>
  <tr>
    <td>searchbp_task_stats_resource_cpu_usage_currentAvg
    </td>

>     <td>The average CPU time for all tasks currently running on the node on the search task level.

Reviewer: ```suggestion
    <td>The average CPU time for all tasks currently running on the node containing the search task level.
```

Reviewer 3: ```suggestion
    <td>The average CPU time for all tasks currently running on the node at the search task level.
```
Here it means the metric is at co-ordinator task level on the node.

    </td>
  </tr>
  <tr>
    <td>searchbp_task_stats_resource_elaspedtime_usage_cancellationCount
    </td>

>     <td>The number of tasks marked for cancellation because of excessive elapsed time since the node last restarted on the search task level.

Reviewer: ```suggestion
    <td>The number of tasks marked for cancellation because of excessive elapsed time since the last restart of the node containing the search task level.
```

    </td>
  </tr>
  <tr>
    <td>searchbp_task_stats_resource_elaspedtime_usage_currentMax
    </td>

>     <td>The maximum elapsed time for all tasks currently running on the node on the search task level.

Reviewer: ```suggestion
    <td>The maximum elapsed time for all tasks currently running on the node containing the search task level.
```

    </td>
  </tr>
  <tr>
    <td>searchbp_task_stats_resource_elaspedtime_usage_currentAvg
    </td>

>     <td>The average elapsed time for all tasks currently running on the node on the search task level.

Reviewer: ```suggestion
    <td>The average elapsed time for all tasks currently running on the node containing the search task level.
```

    </td>
  </tr>
 </tbody>
</table> 

## Dimensions reference

| Dimension            | Return values                                   |
|----------------------|-------------------------------------------------|
| `ShardID`              | The ID of the shard, for example, `1`.           |
| `IndexName`            | The name of the index, for example, `my-index`.   |
| `Operation`            | The type of operation, for example, `shardbulk`.  |
| `ShardRole`            | The shard role, for example, `primary` or `replica`.                            |
| `Exception`            | OpenSearch exceptions, for example, `org.opensearch.index_not_found_exception`. |
| `Indices`              | The list of indexes in the request URL.        |

> | `HTTPRespCode`         | The response code from OpenSearch, for example, `200`. |

Reviewer 2: "The OpenSearch response code"?

> | `HTTPRespCode`         | The response code from OpenSearch, for example, `200`. |

Reviewer: ```suggestion
| `HTTPRespCode`         | The OpenSearch response code, for example, `200`. |
```

   </tbody>
</table>

## Relevant dimensions: `NodeID`, `searchbp_mode`

<table>
  <thead style="text-align: left">
  <tr>
    <th>Metric</th>
    <th>Description</th>
  </tr>
 </thead>
 <tbody> 
    <tr>
    <td>searchbp_shard_stats_cancellationCount
    </td>
    <td>The number of tasks marked for cancellation on the shard task.
    </td>
  </tr>
  <tr>
    <td>searchbp_shard_stats_limitReachedCount
    </td>
    <td>The number of times when the cancellable task total exceeded the set cancellation threshold on the shard task.
    </td>
  </tr>
  <tr>
    <td>searchbp_shard_stats_resource_heap_usage_cancellationCount
    </td>
    <td>The number of tasks marked for cancellation because of excessive heap usage since the node last restarted on the shard task.
    </td>
  </tr>
  <tr>
    <td>searchbp_shard_stats_resource_heap_usage_currentMax
    </td>
    <td>The maximum heap usage for tasks currently running on the shard task.
    </td>
  </tr>
  <tr>
    <td>searchbp_shard_stats_resource_heap_usage_rollingAvg
    </td>
    <td> The rolling average heap usage for the _n_ most recent tasks on the shard task. The default value for _n_ is 100.
    </td>
  </tr>
  <tr>

>     <td>searchbp_shard_stats_resource_cpu_usage_cancellationCount

Reviewer 4: @kaushalmahi12 : We follow the `CamelCase` naming convention for metric naming. Can you provide the same for these metrics.

Author: > @kaushalmahi12 : We follow the `CamelCase` naming convention for metric naming. Can you provide the same for these metrics.

These look right to me. Can you please indicate which ones are incorrect?

Reviewer 4: For reference https://github.com/opensearch-project/documentation-website/blob/9b354eb15e519b33ea702f8e7cc67bde1aa4fb6e/_monitoring-your-cluster/pa/reference.md?plain=1#L30 

`CPU_Utilization` - The Metric name begins with Uppercase letter, and uses `_` .  Following on same line, `searchbp_shard_stats_resource_cpu_usage_cancellationCount` will be `SearchBP_Shard_Stats_Resource_CPU_Usage_CancellationCount`.

Author: Thanks, @khushbr ! @Naarcha-AWS will do the doc review and merge.
